go-controller/pkg/ovn/loadbalancer.go

```diff
@@ -188,10 +188,24 @@ func (ovn *Controller) createLoadBalancerRejectACL(lb string, sourceIP string, s
 	vip := util.JoinHostPortInt32(sourceIP, sourcePort)
 	// NOTE: doesn't use vip, to avoid having brackets in the name with IPv6
 	aclName := fmt.Sprintf("%s-%s:%d", lb, sourceIP, sourcePort)
+	aclName = strings.ReplaceAll(aclName, ":", "\\:")
+	// ACL names are limited to 63 characters
+	if len(aclName) > 63 {
+		// Add the length of the IP (max 39 with colons), plus 14 for '\\' for each ':' in IP,
+		// plus length of sourcePort (max 5 char), plus 3 for additional '\\:' to separate,
+		// plus '-' between lb and IP. With full IPv6 address and 5 char port, max ipPortLen is 62.
+		srcPortStr := fmt.Sprintf("%d", sourcePort)
+		ipPortLen := len(sourceIP) + 14 + len(srcPortStr) + 3 + 1
```

Comment:
this buffer logic is only for ipv6 right? so we need to check if the ip is ipv6 or not and calculate right max buffer for ipv4?

Reply:
Done. In my head wouldn't hit in IPv4. But I added a check for IPv4/IPv6 and adjusted comment and lengths for each.